python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/ecs.py

```diff
@@ -0,0 +1,251 @@
+                            "environment": environment_overrides,
+                        }
+                    )
+
+            params: Dict[str, Any] = {
+                "taskDefinition": task_definition,
+                "count": count,
+                "launchType": launch_type,
+                "cluster": cluster,
+                "group": group,
+                "networkConfiguration": network_configuration,
+                "overrides": overrides,
+                "placementConstraints": placement_constraints,
+                "placementStrategy": placement_strategy,
+                "platformVersion": platform_version,
+                "referenceId": reference_id,
+                "startedBy": started_by,
+                "tags": tags,
+                "clientToken": client_token,
+                "volumeConfigurations": volume_configurations,
+                "capacityProviderStrategy": capacity_provider_strategy,
+                "enableECSManagedTags": enable_ecs_managed_tags,
+                "enableExecuteCommand": enable_execute_command,
+                "propagateTags": propagate_tags,
+            }
+
+            params = {
+                k: v for k, v in params.items() if v is not None
+            }  # boto3 doesn't use None as default values
+
+            response = self._client.run_task(**params)
+
+            tasks: List[str] = [task["taskArn"] for task in response["tasks"]]
+
+            try:
+                response = self._wait_for_tasks_completion(tasks=tasks, cluster=cluster)
+
+                # collect logs from all containers
+                for task in response["tasks"]:
+                    task_id = task["taskArn"].split("/")[-1]
+
+                    for container in task["containers"]:
+                        if log_config := log_configurations.get(container["name"]):
+                            if log_config["logDriver"] == "awslogs":
+                                log_group = log_config["options"]["awslogs-group"]
+
+                                # stream name is combined from: prefix, container name, task id
+                                log_stream = f"{log_config['options']['awslogs-stream-prefix']}/{container['name']}/{task_id}"
+
+                                if isinstance(self._message_reader, PipesCloudWatchMessageReader):
+                                    self._message_reader.consume_cloudwatch_logs(
+                                        log_group,
+                                        log_stream,
+                                        start_time=int(task["createdAt"].timestamp() * 1000),
+                                    )
+                            else:
+                                context.log.warning(
+                                    f"[pipes] Unsupported log driver {log_config['logDriver']} for container {container['name']} in task {task['taskArn']}. Dagster Pipes won't be able to receive messages from this container."
```

Comment:
@schrockn I Pipes emitting messages starting with `[pipes]` (giving an example here, will delete later) - what's the purpose of this? Is it supposed to be used only in pipes internals?

Reply:
I'll try to add them to all our logging inside Pipes clients then